fendermint/storage/src/im.rs

```diff
@@ -120,7 +120,7 @@ pub struct Transaction<'a, S: KVStore, M> {
     _mode: M,
 }
 
-impl<'a, S: KVStore> KVTransaction for Transaction<'a, S, Write> {
+impl<S: KVStore> KVTransaction for Transaction<'_, S, Write> {
```

Comment:
You'll see a few places were I removed lifetime specifiers that are able to be elided.  I tried a few different toolchains when I was looking for a way to get all the dependencies to work together when published to crates (i.e. without a lock file).  One of the versions of `clippy` complained that these can be elided and should be removed.  
I ended up finding a way to stick with the current `1.81.0` toolchain, but afaict these lifetime specifiers are not needed, so I removed them.